dialogue-target/src/main/java/com/palantir/dialogue/Response.java

```diff
@@ -46,4 +46,6 @@
      */
     @Override
     void close();
+
+    <T> Optional<T> getTag(TagKey<T> tagKey);
```

Comment:
this is possibly a non-ideal api, beacuse it can only really be implemented by if/else chains. OTOH, if the if/else chain is more than a couple deep, we have a problem.